client/js/lounge.js

```diff
@@ -1344,4 +1367,31 @@ $(function() {
 			}
 		}
 	);
+
+	window.addEventListener(
+		"popstate",
+		function(e) {
+			var state = e.state;
+			if (!state) {
+				return;
+			}
+			var target;
+			switch (state.view) {
```

Comment:
There should be a way to get rid of this switch and simplify the logic. Perhaps if channelId is not null then find the channel, otherwise try to find a window (without hard coding known windows).